app/src/main/java/emu/skyline/MainViewModel.kt

```diff
@@ -73,4 +73,20 @@ class MainViewModel @Inject constructor(@ApplicationContext context : Context, p
             }
         }
     }
+
+    /**
+     * This checks if the roms have changed since the last time they were loaded and if so it reloads them
+     */
+    fun checkRomHash(context : Context, searchLocation : Uri, systemLanguage : Int) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
You should check that state is `MainState.Loaded` here to avoid running the check if it's still loading or there was an error. Do a check for `state != MainState.Loaded` and return early in that case.

Reply:
I just tested this PR, without that check the app just crashes outright if the app is being installed from scratch.

Reply:
👍